crates/prover-alloy/src/lib.rs

```diff
@@ -31,6 +30,15 @@ type AlloyFillProvider = FillProvider<
     Ethereum,
 >;
 
+#[async_trait]
+#[cfg_attr(feature = "testutils", mockall::automock)]
+pub trait Provider {
```

Comment:
Just not sure about the trait name - could be confused with actual `alloy::providers::Provider`. Maybe `RpcProvider`?

Reply:
The goal is to not expose `alloy::providers::Provider` at some point, maybe we should rename this crate to something else :)

Reply:
@Freyskeyd I know we will not expose `alloy::providers::Provider`, we have this crate to hide it behind. Just using the same trait name `Provider` could be confusing for somebody looking at the code where it is used.